### PR TITLE
Delete dead/unreachable code related to remoting

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
(There may a few deleted pieces that are not strictly related to remoting, but there were next to the remoting code that I was deleting.)

File: src/inc/MSCOREE.IDL [hunk 1]
```diff
@@ -40,13 +40,6 @@ cpp_quote("EXTERN_GUID(CLSID_ComCallUnmarshal, 0x3F281000,0xE95A,0x11d2,0x88,0x6
 cpp_quote("EXTERN_GUID(CLSID_ComCallUnmarshalV4, 0x45fb4600,0xe6e8,0x4928,0xb2,0x5e,0x50,0x47,0x6f,0xf7,0x94,0x25);")
 #endif // FEATURE_COMINTEROP
 
-
-#ifdef FEATURE_COMINTEROP
-// IID IManagedObject : uuid(C3FCC19E-A970-11d2-8B5A-00A0C9B7C9C4)
-cpp_quote("EXTERN_GUID(IID_IManagedObject, 0xc3fcc19e, 0xa970, 0x11d2, 0x8b, 0x5a, 0x00, 0xa0, 0xc9, 0xb7, 0xc9, 0xc4);")
```
Comment (AaronRobinsonMSFT) [inline src/inc/MSCOREE.IDL]: Thank you! This was on my list of things to remove :)

File: src/vm/classnames.h [hunk 2]
```diff
@@ -93,19 +93,13 @@
 #define g_ExceptionClassName "System.Exception"
 #define g_ExecutionEngineExceptionClassName "System.ExecutionEngineException"
 
-#define g_MarshalByRefObjectClassName "System.MarshalByRefObject"
```
Comment (AaronRobinsonMSFT) [inline src/vm/classnames.h]: I think we may have a CoreFX test for this?

Comment (jkotas) [reply src/vm/classnames.h]: Yes, this MarshalByRefObject exists in public surface. It is just a dummy type for backward compatibility, not treated in any special way by the runtime.

**Comment by jkotas** [comment]
@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test
@dotnet-bot test Ubuntu arm Cross Checked crossgen_comparison Build and Test
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test

**Pull request merged**